engine/storage/src/org/apache/cloudstack/storage/image/db/SnapshotDataStoreDaoImpl.java

```diff
@@ -272,22 +294,18 @@ public SnapshotDataStoreVO findOldestSnapshotForVolume(Long volumeId, DataStoreR
     @Override
     @DB
     public SnapshotDataStoreVO findParent(DataStoreRole role, Long storeId, Long volumeId) {
-        TransactionLegacy txn = TransactionLegacy.currentTxn();
-        try (
-                PreparedStatement pstmt = txn.prepareStatement(parentSearch);
-            ){
-            pstmt.setLong(1, storeId);
-            pstmt.setString(2, role.toString());
-            pstmt.setLong(3, volumeId);
-            try (ResultSet rs = pstmt.executeQuery();) {
-                while (rs.next()) {
-                    long sid = rs.getLong(1);
-                    long snid = rs.getLong(3);
-                    return findByStoreSnapshot(role, sid, snid);
-                }
-            }
-        } catch (SQLException e) {
-            s_logger.debug("Failed to find parent snapshot: " + e.toString());
+
+        SearchCriteria<SnapshotDataStoreVO> sc = parentSnapshotSearch.create();
+        sc.setParameters("volume_id", volumeId);
+        sc.setParameters("store_role", role.toString());
+        sc.setParameters("state", ObjectInDataStoreStateMachine.State.Ready.name());
+        sc.setParameters("store_id", storeId);
+        sc.setJoinParameters("snapshotVOSearch", "hypervisorType", Hypervisor.HypervisorType.XenServer);
```

Comment:
@niteshsarda Can you confirm if XenServer is the only supported hypervisor with valid parent snapshot ID for a given snapshot ID?
Instead of hardcoding the hypervisor type, could we fetch hypervisor type categorically?

Reply:
@sateesh-chodapuneedi : Verified that parent snapshot ID is only required in case of XenServer.
Also, as per your suggestion changed the code to fetch hypervisor type categorically.

Please review latest code.